src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,110 @@
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+import React, { useCallback, useMemo, useState } from 'react';
+
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+import { Gender } from '../types/Gender';
+
+type Props = {
+  onSelect: (person: Person) => void,
+  delay: number,
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  onSelect,
+  delay,
+}) => {
+  const [visibleQuery, setVisibleQuery] = useState('');
+  const [inputQuery, setInputQuery] = useState('');
+  const [isListVisible, setIsListVisible] = useState(false);
+
+  const currentInputQuery = useCallback(
```

Comment:
The problem was that you pass debounce directly, but useCallback/useMemo requires a callback instead.
https://stackoverflow.com/questions/69830440/react-hook-usecallback-received-a-function-whose-dependencies-are-unknown-pass